lib/export/coldp/files/taxon.rb

```diff
@@ -130,6 +128,15 @@ def self.generate(otus, root_otu_id = nil, reference_csv = nil, prefer_unlabelle
         remarks
       }
 
+      if root_otu_id.nil?
+        Current.project_id = otus[0].project_id
+      else
+        Current.project_id = Otu.find(root_otu_id).project_id
```

Comment:
Just use `project_id: otu.porject_id` as needed.  In general don't touch Current.